IDEMSInternational/parenting-app-ui
# fix: standardise newline char in parsed json outputs

PR Checklist


## Description

As documented in issue #1838, some strings that appear in the json files generated from the templates would contain `\r\n` (CRLF) newline characters, as opposed to the standard `\n` (LF). This causes an issue when the strings are sent to be translated as "\r" characters would appear, unparsed, in the strings which could cause confusion for the translators.

This PR enforces the `\n` newline character by replacing all instances of `\r\n` with `\n` in all the outputted app-data sheets.

## Testing notes

1. Checkout `master` and pull the latest changes. 
2. Checkout `fix/standardise-newlines-in-parser`
3. Run `yarn workflow sync_sheets`
4. In VSCode's Source Control, you should see only changes that have been made to content since the last sync to master, alongside the newline character changes enforced by this PR (you may need to stage and un-stage all changes to remove erroneous file differences). At the time of writing, those changes are as shown in the screenshot below. The only file from PLH that previously had `\r\n` line endings is `packages/app-data/sheets/data_list/survey_question.json`. You should see by comparing the differences that these have been replaced with `\n`.

<img width="1000" alt="Screenshot 2023-03-14 at 16 26 55" src="https://user-images.githubusercontent.com/64838927/225075271-3505168b-b99c-4b60-bea5-651f3d3aaedc.png">

It would be sensible to check that this content still appears in the app as expected.

The sheets for the WASH App contain many more examples of the `\r\n` character (and it was these that originally prompted the issue to be raised). However, as this feature branch was created from the master branch, it is more difficult to compare the results of the updated AppDataConverter on the WASH sheets. I have tested this and can confirm that the only changes appear to be the correct substitution of `\n` for `\r\n` in those files.

## Git Issues

Closes #1838

## Screenshots/Videos

The changes made by the new AppDataConverter to the file referenced in the original issue:
<img width="1000" alt="Screenshot 2023-03-14 at 15 52 05" src="https://user-images.githubusercontent.com/64838927/225078490-72dc35c2-0ae1-409d-8746-c389e04390ae.png">
